core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -100,7 +100,17 @@ private[spark] class DiskBlockManager(conf: SparkConf, deleteFilesOnStop: Boolea
 
   /** List all the blocks currently stored on disk by the disk manager. */
   def getAllBlocks(): Seq[BlockId] = {
-    getAllFiles().map(f => BlockId(f.getName))
+    getAllFiles().flatMap { f =>
+      try {
+        Some(BlockId(f.getName))
+      } catch {
+        case _: UnrecognizedBlockId =>
+          // This does not handle a special-case of a temporary file
+          // created by [[SortShuffleWriter]].
+          log.warn(s"Encountered an unexpected file in a managed directory: $f")
```

Comment:
My question here: is it really something we need to warn users? What should they do about it? Here we are getting block ids from file names, so it's legal to meet a temporary file at this place.

Reply:
I am not convinced we should log either. I've added this line because it was suggested by @jerryshao.